controllers/clustermesh/clustermesh_controller.go

```diff
@@ -181,12 +183,91 @@ func (r *ClusterMeshReconciler) reconcileNormal(ctx context.Context, cluster *cl
 		return ctrl.Result{}, err
 	}
 
-	return r.ReconcileRoutesFactory(r, ctx, clSpec)
+	result, err := r.ReconcileRoutesFactory(r, ctx, clSpec)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	err = r.validateClusterMesh(ctx, clustermesh)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return result, nil
+}
+
+func (r *ClusterMeshReconciler) validateClusterMesh(ctx context.Context, clustermesh *clustermeshv1beta1.ClusterMesh) error {
+	vpcReady, routesReady, sgReady := true, true, true
+
+	vpcPeeringConnections, err := crossplane.GetOwnedVPCPeeringConnections(ctx, clustermesh, r.Client)
+	if err != nil {
+		return err
+	}
+
+	if len(vpcPeeringConnections.Items) == 0 {
+		vpcReady, routesReady = false, false
+	}
+
+	for _, vpcPeeringConnection := range vpcPeeringConnections.Items {
+		if !checkConditionsReadyAndSynced(vpcPeeringConnection.Status.Conditions) {
+			vpcReady = false
+		}
+
+		routes, err := crossplane.GetOwnedRoutes(ctx, &vpcPeeringConnection, r.Client)
+		if err != nil {
+			return err
+		}
+
+		if len(routes.Items) == 0 {
```

Comment:
same as above, but it may be trickier to calculate the desired number of routes since it depends on the number of rtbs 🤔